### DISCUSSION TITLE
Is there a way to host Pocketbase db to Vercel or any other free platforms?

### DISCUSSION THREAD

**Discussion by albeniskerqeli10** [pocketbase/pocketbase#3560] [Q&A] [answered]
I'm using Pockethost to host my Pocketbase database, but I noticed that it has a slow response when handling frontend requests. So, are there any free, better alternatives like Vercel or other platforms to host Pocketbase?

**Comment by ganigeorgiev** [comment, marked as answer]
Other free options that are commonly used are [Hop.io](https://github.com/pocketbase/pocketbase/discussions/2856) or [Fly.io](https://github.com/pocketbase/pocketbase/discussions/537).

_Side-note: If you use Vercel, then I suspect that you use also Next.js SSR or RSC. A lot of developers seem to jump straight to it, but my general recommendation would be, if you are not working on a website, to NOT use JS SSR/RSC and prefer client-side rendering (eg. SPA). If you use JS SSR/RSC you no longer have "just frontend".
If that's the case, the slow response times could be caused by the different locations of the servers because you are probably proxy-ing every client request from your node server to PocketBase (aka. browser->node->PocketBase). PocketBase comes with its own [API rules](https://pocketbase.io/docs/api-rules-and-filters/) to manage the access to your data and it is safe to make direct client-side request against it. Prefer the node server only for one-off admin actions (eg. captcha verification, Stripe webhook handling, etc.) to limit the excessive requests between the services.
Another possible cause could be also just some resources restriction/temp unavailability of the PocketHost server (you may want to check [their repo](https://github.com/pockethost/pockethost/discussions) to see if someone else has reported something)._

**Reply by albeniskerqeli10** [reply]
No, I'm using Vercel just for the frontend (React + Vite), and I'm not using Next.js at all. 

However, I will explore these two options for Pocketbase to see if the issue truly lies in the database hosting.

**Reply by RayyanNafees** [reply]
Fly.io is no longer free

They gone down the same path as Heroku did

**Reply by ganigeorgiev** [reply]
@RayyanNafees I haven't personally tested to see what has changed but I think it is still free. Based on their [Pricing docs](https://fly.io/docs/about/pricing/#plans) new users starts with $5 credit but if your usage doesn't exceed the free allowance this credit will be never used:

> The free trial credit doesn’t expire.
> Usage beyond our [free resource allowances](https://fly.io/docs/about/pricing/#free-allowances) consumes your free trial credit.

It seems that now all new registrations require a credit card (previously it was required only for specific regions).

> All plans require a [credit card](https://fly.io/docs/about/billing/#payment-options) on file

**Reply by RayyanNafees** [reply]
Though it works good with Cloudflare Pages & Render for free

**Comment by WombleWoo7547** [comment]
Use Cloudflare tunnels, buy a rpi 0, install docker and plug it into a spare usb socket and call it a day. Easy $10 one-time payment and after 2 months pays itself off instead of using cloud resources ($5 a month "nanode" on akamai).

**Comment by CharanKuberns** [comment]
The slow response on PocketHost is a known issue on the free tier. Shared resources and occasional cold starts on that plan cause exactly the latency you are describing.

The important thing to understand first is that PocketBase cannot run on Vercel. Vercel is a serverless platform and PocketBase requires a single long-running persistent process with persistent storage. Serverless functions get created and destroyed per request, which is fundamentally incompatible with how PocketBase works. So your current split setup (Vercel for frontend, separate host for PocketBase) is the right architecture. You just need a better host on the PocketBase side.

**Options worth trying:**
Fly.io is the most recommended in the PocketBase community specifically. You get persistent VMs, persistent volumes for your data, and no cold starts. The binary runs exactly as intended as a long-lived process. Performance is noticeably better than PocketHost. The setup involves the Fly CLI but their PocketBase deployment guide is well documented.

Railway is another solid option if you want slightly easier setup than Fly.io. Persistent containers with disk, and usage-based billing that is transparent and predictable.

Kuberns is worth a look if you want to avoid the manual setup entirely. It is an agentic AI deployment platform where you connect your GitHub repo and an AI agent handles the full deployment automatically including infrastructure, environment, and persistent storage configuration. No config files, no manual setup. The agent figures out what your stack needs and deploys it. Given that you are dealing with a split frontend/backend setup, having the deployment handled automatically removes a lot of the friction.

Either way the key requirement is a persistent process with persistent storage, which all three options above satisfy. Fly.io or Railway are solid free/low-cost choices if you want to self-manage. Kuberns if you want it handled for you.

**Reply by ganigeorgiev** [reply]
@CharanKuberns AI generated responses are not welcomed here.

**Comment by awbrine-collab** [comment]
Just putting another option out there (tho not free but cheap) use Coolify (free selfhost alternative to things like vercel) Or use there cloud version 

Then you can deploy all ur projects there

**Comment by awbrine-collab** [comment]
And also another tip if using react and vite embed the front end into pb then only one exe for backend and frontend

**Comment by mt-shihab26** [comment]
I am hosting my application on vercel. Using edge function proxy with embedded vite application on a go project with pocketbase. You can checkout this https://github.com/mt-shihab26/urlz
